Aquality.Selenium/src/Aquality.Selenium/Browsers/BrowserTabNavigation.cs

```diff
@@ -0,0 +1,84 @@
+﻿using Aquality.Selenium.Core.Localization;
+using OpenQA.Selenium.Remote;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Aquality.Selenium.Browsers
+{
+    public class BrowserTabNavigation : IBrowserTabNavigation
+    {
+        private readonly RemoteWebDriver driver;
+
+        internal BrowserTabNavigation(RemoteWebDriver driver)
+        {
+            this.driver = driver;
+        }
+
+        private ILocalizedLogger Logger => AqualityServices.LocalizedLogger;
+
+        public void CloseTab()
+        {
+            Logger.Info("loc.browser.tab.close");
+            driver.Close();
+        }
+
+        public string GetTabHandle()
+        {
+            Logger.Info("loc.browser.get.tab.handle");
+            return driver.CurrentWindowHandle;
+        }
+
+        public IList<string> GetTabHandles()
+        {
+            Logger.Info("loc.browser.get.tab.handles");
+            return driver.WindowHandles;
+        }
+
+        public void OpenNewTab(bool switchToNew = true)
+        {
+            Logger.Info("loc.browser.tab.open.new");
+            AqualityServices.Browser.ExecuteScript(JavaScript.OpenNewTab);
+            if (switchToNew)
+            {
+                SwitchToNewTab();
+            }
+        }
+
+        public void SwitchToNewTab(bool closeCurrent = false)
+        {
+            Logger.Info("loc.browser.switch.to.new.tab");
+            var newTab = GetTabHandles().Last();
+            CloseAndSwitch(newTab, closeCurrent);
+        }
+
+        public void SwitchToTab(string handle, bool closeCurrent = false)
+        {
+            Logger.Info("loc.browser.switch.to.tab.handle", handle);
+            CloseAndSwitch(handle, closeCurrent);
+        }
+
+        public void SwitchToTab(int index, bool closeCurrent = false)
+        {
+            Logger.Info("loc.browser.switch.to.tab.index", index);
+            var names = GetTabHandles();
+            if (index >= names.Count || index < 0)
```

Comment:
Please change to 
index < 0 || names.Count <= index - this is more readable when you organize values from the lowest to the highest

Reply:
fixed